File: scripts/vendor_check.py
#!/usr/bin/env python3
import os
import sys

ROOT = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
VENDOR = os.path.join(ROOT, 'vendor')
FILES = ['papaparse.min.js', 'chart.umd.min.js', 'tabulator.min.css', 'tabulator.min.js']

missing = [f for f in FILES if not os.path.isfile(os.path.join(VENDOR, f))]
if missing:
    print('Missing vendor files:', ', '.join(missing))
    sys.exit(1)
print('All vendor files present.')

File: scripts/vendor-check.js
import { existsSync } from 'fs';
import { join } from 'path';

const vendorFiles = [
  'papaparse.min.js',
  'chart.umd.min.js',
  'tabulator.min.css',
  'tabulator.min.js',
];

let failed = false;
for (const f of vendorFiles) {
  const p = join('vendor', f);
  if (!existsSync(p)) {
    console.error('Missing vendor file:', p);
    failed = true;
  }
}
if (failed) process.exit(1);
console.log('All vendor files present.');
